### PR TITLE
feat: officially expose testing helpers

### LINKED ISSUE #4930: feat: Publicly Expose `fixture<T>` in `@microsoft/fast-tooling`
# 🙋 Feature Request

`@microsoft/fast-tooling` has [`fixture<T>`](https://github.com/microsoft/fast/blob/e85551b6f4c3c79f5c5d8500a0ed2b5988a8def6/packages/tooling/fast-tooling/src/web-components/fixture.ts#L110) which seems like a really useful helper function for tests. It's not currently publicly exposed, can that be changed?

## 🤔 Expected Behavior

Allow us to import [`fixture<T>`](https://github.com/microsoft/fast/blob/e85551b6f4c3c79f5c5d8500a0ed2b5988a8def6/packages/tooling/fast-tooling/src/web-components/fixture.ts#L110).

## 😯 Current Behavior

[`fixture<T>`](https://github.com/microsoft/fast/blob/e85551b6f4c3c79f5c5d8500a0ed2b5988a8def6/packages/tooling/fast-tooling/src/web-components/fixture.ts#L110) is not publicly exposed, so can't import it.

## 💁 Possible Solution

Add it to the `@microsoft/fast-tooling` package.

### PR DESCRIPTION
**Pull request by EisenbergEffect**
# Pull Request

## 📖 Description

This PR fixes up the organization of the testing helpers in foundation and exposes them through the `exports` configuration of the `package.json`.

### 🎫 Issues

* Closes #4930

## 👩‍💻 Reviewer Notes

@nicholasrice Can you check that I've handled the `package.json` correctly? I didn't rename the `index.ts` file for the main package. I just tried to fix up the testing helpers a bit based on what you were doing in the ssr package.

## 📑 Test Plan

No changes to functionality. Existing tests, including those that use these helpers, still pass.

### General

- [x] I have included a change request file using `$ yarn change`
- [ ] I have added tests for my changes.
- [x] I have tested my changes.
- [x] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/microsoft/fast/blob/master/CONTRIBUTING.md) documentation and followed the [standards](/docs/community/code-of-conduct/#our-standards) for this project.

## ⏭ Next Steps

We should probably follow this up with a change to adopt the "exports" naming scheme and set up all the `package.json` files properly to use this feature. We may want to leverage a development and production export configuration for `fast-element` as well.

### CODE REVIEW

**Comment by EisenbergEffect** [comment]
@nicholasrice Any idea why TS can't find the declaration file now? I just changed the folder and file name. I did see that the `exports` package.json feature requires TS 4.5, but I'm not actually using that for the import here. Thoughts?

File: packages/web-components/fast-foundation/package.json [hunk 1]
```diff
@@ -19,6 +19,10 @@
   "types": "dist/fast-foundation.d.ts",
   "unpkg": "dist/fast-foundation.min.js",
   "type": "module",
+  "exports": {
+    "." : "./dist/esm/index.js",
+    "./testing": "./dist/esm/testing/exports.js"
```
Comment (nicholasrice) [inline packages/web-components/fast-foundation/package.json:24]: Nice, I really like this feature

**Review by nicholasrice — APPROVED** [review]

**Comment by nicholasrice** [comment]
> @nicholasrice Any idea why TS can't find the declaration file now? I just changed the folder and file name. I did see that the `exports` package.json feature requires TS 4.5, but I'm not actually using that for the import here. Thoughts?

I think this is because in the react-wrapper package you are importing from a path that isn't enumerated in the `"exports"` field of the fast-foundation/package.json (`@microsoft/fast-foundation/dist/esm/testing/exports`). That should really import from `@microsoft/fast-foudation/testing`, however I think you'll run into an issue w/ that as well because of our TypeScript version. This may be blocked until we can upgrade TypeScript

**Comment by EisenbergEffect** [comment]
> > @nicholasrice Any idea why TS can't find the declaration file now? I just changed the folder and file name. I did see that the `exports` package.json feature requires TS 4.5, but I'm not actually using that for the import here. Thoughts?
> 
> I think this is because in the react-wrapper package you are importing from a path that isn't enumerated in the `"exports"` field of the fast-foundation/package.json (`@microsoft/fast-foundation/dist/esm/testing/exports`). That should really import from `@microsoft/fast-foudation/testing`, however I think you'll run into an issue w/ that as well because of our TypeScript version. This may be blocked until we can upgrade TypeScript

I wondered if it was something like that...this can easily wait to merge until we can update that. I just want to make sure we've got all this patched up along with the TS version before we start publishing beta packages.

**Comment by nicholasrice** [comment]
Yeah if the exports field is enumerated, Node will prevent import drilling into the underlying file structure, which is actually pretty nifty, it provides a lot of control over the package structure and helps prevent access to code intended to be private. But it definitely breaks older versions of TypeScript.

**Comment by EisenbergEffect** [comment]
Superseded by #5958

**Pull request closed without merging**